tests/cli_test.py

```diff
@@ -56,6 +57,203 @@ def cli11_optional_option_arg_test():
                                       '-c', chain, '-k', key, '{}'])
     assert(b'signatures' in output)
 
+    """Test option --abi-file """
+    token_abi_path  = os.path.abspath(os.getcwd() + '/../unittests/contracts/eosio.token/eosio.token.abi')
+    system_abi_path = os.path.abspath(os.getcwd() + '/../unittests/contracts/eosio.system/eosio.system.abi')
+    token_abi_file_arg = 'eosio.token' + ':' + token_abi_path
+    system_abi_file_arg = 'eosio' + ':' + system_abi_path
+
+    # no option --abi-file
+    account = 'eosio.token'
+    action = 'transfer'
+    unpacked_action_data = '{"from":"aaa","to":"bbb","quantity":"10.0000 SYS","memo":"hello"}'
+    cmd = ['./programs/cleos/cleos', 'convert', 'pack_action_data', account, action, unpacked_action_data]
+    outs=None
+    errs=None
+    try:
+        popen=subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        outs,errs=popen.communicate()
+        popen.wait()
+    except subprocess.CalledProcessError as ex:
+        print(ex.output)
+    assert(b'Failed to connect to nodeos' in errs)
+
+    # invalid option --abi-file
+    invalid_abi_arg = 'eosio.token' + ' ' + token_abi_path
+    cmd = ['./programs/cleos/cleos', '--abi-file', invalid_abi_arg, 'convert', 'pack_action_data', account, action, unpacked_action_data]
+    outs=None
+    errs=None
+    try:
+        popen=subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        outs,errs=popen.communicate()
+        popen.wait()
+    except subprocess.CalledProcessError as ex:
+        print(ex.output)
+    assert(b'please specify --abi-file in form of <contract name>:<abi file path>.' in errs)
+
+    # pack token transfer data
+    account = 'eosio.token'
+    action = 'transfer'
+    unpacked_action_data = '{"from":"aaa","to":"bbb","quantity":"10.0000 SYS","memo":"hello"}'
+    packed_action_data = '0000000000008c31000000000000ce39a08601000000000004535953000000000568656c6c6f'
+    cmd = ['./programs/cleos/cleos', '--abi-file', token_abi_file_arg, 'convert', 'pack_action_data', account, action, unpacked_action_data]
```

Comment:
The purpose of the new option --abi-file is to provide a capability to do conversion offline. To make sure not use local Nodeos, use a non-existing port in "-u" option of cleos like "cleos -u 127.0.0:12345".

Reply:
Fixed.